### ISSUE TITLE
inference DoRA of Advanced_diffusion_training issues

### ISSUE THREAD

**Issue by SlZeroth** [huggingface/diffusers#7422]
### What does this PR do?

this is the command line for training LORA + DORA 

`accelerate launch --num_cpu_threads_per_process=2 C:/Users/qkrwn/ML/diffusers/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py --pretrained_model_name_or_path="streamize/xxmixgirl" --instance_prompt="a photo of a ohwx woman" --validation_prompt="a photo of a ohwx woman in the home" --output_dir="lora-rank100-2000-20" --resolution=1024 --train_batch_size=1 --repeats=1 --gradient_accumulation_steps=1 --gradient_checkpointing --learning_rate=1.0 --text_encoder_lr=1.0 --optimizer="prodigy" --mixed_precision="fp16" --train_text_encoder_ti --lr_scheduler=constant --lr_warmup_steps=0 --rank=100 --use_dora --max_train_steps=2000 --checkpointing_steps=2000 --seed=1234 --instance_data_dir="C:/Users/qkrwn/OneDrive/face_images/bad_people/young"`

this error message is shown when training is finshed

`Loading unet.
Loading adapter weights from state_dict led to unexpected keys not found in the model:  ['down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn1.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn1.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn1.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn1.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn2.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn2.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn2.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn2.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_blocks.0.attn1.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_blocks.0.attn1.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_blocks.0.attn1.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_blocks.0.attn1.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_blocks.0.attn2.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_blocks.0.attn2.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_blocks.0.attn2.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_b .....`

An error message occurs at the end of training, but the trained safetensors file is successfully created. If you perform inference with the created file, the following error occurs.

this is the test inference code for test error

```
from diffusers import StableDiffusionXLPipeline, EulerAncestralDiscreteScheduler

pipeline.scheduler = EulerAncestralDiscreteScheduler.from_config(pipeline.scheduler.config)
pipeline.enable_model_cpu_offload()
pipeline.enable_xformers_memory_efficient_attention()

pipeline.load_lora_weights("./young-r100-lr3e-4.safetensors")
pipeline.fuse_lora(lora_scale=0.85)

images = pipeline(
    "photo of ohwx woman",
    negative_prompt="(worst quality, low quality, illustration, 3d, 2d, painting, cartoons, sketch), tooth, open mouth",
    num_inference_steps=25,
).images
```

console log when inference

CUDA SETUP: Loading binary C:\Users\qkrwn\anaconda3\envs\kohya_ss\lib\site-packages\bitsandbytes\libbitsandbytes_cuda116.dll...
Loading adapter weights from state_dict led to unexpected keys not found in the model:  ['down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.0.attn2.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn1.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn1.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn1.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn1.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn2.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn2.to_k.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn2.to_v.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.0.transformer_blocks.1.attn2.to_out.0.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.transformer_blocks.0.attn1.to_q.lora_magnitude_vector.default_0', 'down_blocks.1.attentions.1.trans

I push the safetensors file for test this PR ( https://huggingface.co/streamize/dora-trained-on-diffusers )

**Comment by sayakpaul** [comment]
Will look into it.

**Comment by sayakpaul** [comment]
@SlZeroth thanks for reporting it. Could you pull in the changes from https://github.com/huggingface/diffusers/pull/7371 and then run it?

I just did a dummy run:

```bash
CUDA_VISIBLE_DEVICES=1 accelerate launch train_dreambooth_lora_sdxl_advanced.py \
  --pretrained_model_name_or_path=$MODEL_NAME \
  --pretrained_vae_model_name_or_path=$VAE_PATH \
  --dataset_name=$DATASET_NAME \
  --instance_prompt="3d icon in the style of TOK" \
  --validation_prompt="a TOK icon of an astronaut riding a horse, in the style of TOK" \
  --output_dir=$OUTPUT_DIR \
  --mixed_precision="fp16" \
  --resolution=1024 \
  --train_batch_size=2 \
  --repeats=1 \
  --report_to="wandb"\
  --gradient_accumulation_steps=1 \
  --gradient_checkpointing \
  --learning_rate=1e-4 \
  --use_8bit_adam \
  --train_text_encoder_ti\
  --train_text_encoder_ti_frac=0.5\
  --snr_gamma=5.0 \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --rank=8 \
  --max_train_steps=200 \
  --seed="0" \
  --push_to_hub
```

It works:

```python
from diffusers import StableDiffusionXLPipeline
import torch

pipeline = StableDiffusionXLPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.bfloat16)
pipeline.to(device="cuda")

pipeline.load_lora_weights("sayakpaul/3d-icon-SDXL-LoRA")
```

Could you give it a try?

**Comment by 2a3b4c** [comment]
you did not train a dora. this script just is a lora training
@sayakpaul

**Comment by sayakpaul** [comment]
@2a3b4c you're right, thanks for spotting. Just updated my comment. The PR works.

**Comment by 2a3b4c** [comment]
@sayakpaul I test the PR, but also get the same error. i have updated my diffusers to 0.26.0 dev. Could you please give me some advice?

**Comment by sayakpaul** [comment]
You need to install diffusers from the PR branch to be able to correctly test it.

**Comment by 2a3b4c** [comment]
Thanks! It works.

**Comment by SlZeroth** [comment]
its fixed. works now

**Comment by 2a3b4c** [comment]
@sayakpaul i suddenly wander that "pipeline.fuse_lora(lora_scale=0.5)" can work?

**Comment by sayakpaul** [comment]
Yeah of course it should.
